src/features/device-types/device-types.ts

```diff
@@ -159,21 +154,27 @@ async function fetchDeviceTypesAndReschedule(): Promise<
 	Dictionary<DeviceTypeInfo>
 > {
 	try {
-		const promise = withRetries(fetchDeviceTypes).then(
-			async (deviceTypeInfo) => {
-				// when the promise gets resolved, cache it
-				deviceTypesCache = promise;
-				return deviceTypeInfo;
-			},
-		);
+		const promise = withRetries(fetchDeviceTypes).then((deviceTypeInfo) => {
+			deviceTypesCache = Promise.resolve(deviceTypeInfo);
```

Comment:
Strictly speaking there's no need to wrap this since it's only ever used in an `await` context anyway